packages/tracing/src/browser/metrics.ts

```diff
@@ -0,0 +1,279 @@
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+        },
+        { once: true },
+      );
+
+      const updateLCP = (entry: PerformanceEntry) => {
+        // Only include an LCP entry if the page wasn't hidden prior to
+        // the entry being dispatched. This typically happens when a page is
+        // loaded in a background tab.
+        if (entry.startTime < firstHiddenTime) {
+          // NOTE: the `startTime` value is a getter that returns the entry's
+          // `renderTime` value, if available, or its `loadTime` value otherwise.
+          // The `renderTime` value may not be available if the element is an image
+          // that's loaded cross-origin without the `Timing-Allow-Origin` header.
+          this._lcp = {
+            // @ts-ignore
+            ...(entry.id && { elementId: entry.id }),
+            // @ts-ignore
+            ...(entry.size && { elementSize: entry.size }),
+            value: entry.startTime,
+          };
+        }
+      };
+
+      // Create a PerformanceObserver that calls `updateLCP` for each entry.
+      const po = new PerformanceObserver(entryList => {
+        entryList.getEntries().forEach(updateLCP);
+      });
+
+      // Observe entries of type `largest-contentful-paint`, including buffered entries,
+      // i.e. entries that occurred before calling `observe()` below.
+      po.observe({
+        buffered: true,
+        // @ts-ignore
+        type: 'largest-contentful-paint',
+      });
+
+      this._forceLCP = () => {
+        po.takeRecords().forEach(updateLCP);
+      };
+    } catch (e) {
+      // Do nothing if the browser doesn't support this API.
+    }
+  }
+
+  public constructor() {
+    if (global && global.performance) {
+      if (global.performance.mark) {
+        global.performance.mark('sentry-tracing-init');
+      }
+
+      this._trackLCP();
+    }
+  }
+
+  /** Add performance related spans to a transaction */
+  public addPerformanceEntires(transaction: Transaction): void {
+    if (!global || !global.performance || !global.performance.getEntries) {
+      // Gatekeeper if performance API not available
+      return;
+    }
+
+    logger.log('[Tracing] Adding & adjusting spans using Performance API');
+
+    // TODO(fixme): depending on the 'op' directly is brittle.
+    if (transaction.op === 'pageload') {
+      // Force any pending records to be dispatched.
+      this._forceLCP();
+      if (this._lcp) {
+        // Set the last observed LCP score.
+        transaction.setData('_sentry_web_vitals', { LCP: this._lcp });
+      }
+    }
+
+    const timeOrigin = msToSec(performance.timeOrigin);
+    let entryScriptSrc: string | undefined;
+
+    if (global.document) {
+      // tslint:disable-next-line: prefer-for-of
+      for (let i = 0; i < document.scripts.length; i++) {
```

Comment:
Any specific reason to ignore `prefer-for-of`?

Reply:
I copied the implementation from what was there before. I assume it is because we want the performance benefit?

Reply:
Don't remember tbh :D